schema/challengeSchema.js

```diff
@@ -38,6 +38,9 @@ const schema = Joi.object().keys({
   isRequired: Joi.bool(),
   name: Joi.string(),
   order: Joi.number(),
+  react: Joi.bool(),
+  reactRedux: Joi.bool(),
```

Comment:
Why are you adding these properties to the schema?

Reply:
evaluateTest uses them to determine if it needs to add react, redux and so on to the test environment.  In order for that to happen, they have to be present in the challenge object (or the validation step will fail) and that means they have to be in the schema.

Reply:
Understood, is there a way to strip them out of the challenges on build? We are trying to keep the package size to a minimum.

Reply:
If I understand correctly, you don't want these properties making their way into the dist folder - since they're not used outside of testing.  That can be done, by modifying the gulpfile, but the build process doesn't (currently) involve validation.  In other words, the schema isn't enforced.

Does it seem reasonable to add filtering and validation to the build process?  That way all the properties that're currently omitted when `learn` reads the challenges instead never make it into the build of `curriculum`.